git_history_of_project/git_history_of_project.md

```diff
@@ -0,0 +1,620 @@
+
+### Quiz: finding a commit
+
+The output from `git log -n 2` is:
+
+```
+$ git log -n 2
+commit 584977a18319f539f3f5ecde6176db2ba0081cd9 (HEAD -> main, origin/main)
+Author: Vlad Dracula <vlad@tran.sylvan.ia>
+Date:   Mon Mar 14 14:59:40 2022 -0400
+
+    Ignore data files and the results folder.
+
+commit fe532b097861acb8bd3d7f221d6ee741249dc8f0
+Author: Vlad Dracula <vlad@tran.sylvan.ia>
+Date:   Mon Mar 14 14:58:17 2022 -0400
+
+    Add concerns about effects of Mars' moons on Wolfman
+```
+
+Which of the following commands would show you the *most recent* commit you made? Select all correct answers.
+
+- [[X]] `git show HEAD`
+- [[ ]] `git show HEAD~1`
+- [[X]] `git show 584977`
+- [[ ]] `git show 081cd9`
+***
+<div class ="answer">
+The *most recent* commit is the current `HEAD` and its commit number is `584977`. Since both refer to the same commit, you can use either.
+
+The command `git show HEAD~1` will show you one checkpoint earlier in your work, while `git show 081cd9` will give you an error since `081cd9` it is not the *first* six digits of a known commit number.
+</div>
+***
+
+<div class = "care">
+The error you get from entering a non-existent commit number starts with the word "fatal."  The word may be scary, but, in this case, all it is saying is that it could not execute the command. You did not harm anything!
+</div>
+
+### Technical details for the intrigued
+
+<div class = "learnmore">
+When you commit code to Git, it doesn't create a brand new copy of your files from scratch. Instead it records the differences between the version of each file that it already has stored, and the new version.
+
+You could think of each commit not as a checkpoint, but the instructions on how to get to the next checkpoint. The entire repository can be built by following a series of instructions (commits) in order. Similarly you can determine the precise state of an earlier version by following the instructions backwards.
+
+![The three white boxes are shown with a large green arrow pointing from each box to the box above it. The top box is labeled "Last committed version of repository" and the arrow pointing to it from the middle box is labeled "HEAD." The second box is labeled "Next to last version of repository" and the arrow pointing to it from the box below is labeled "HEAD~1." The bottom box is labeled "Previous version of repository" and the arrow pointing to it is labeled "HEAD~2." There are three dots under the boxes, and another three dots under the arrows indicating that this pattern continues.](media/Commits_are_pointers.svg)
+</div>
+
+## Comparing files to prior commits
+
+In order to have some more commits to compare, let's add a few more lines to Dracula's notes about Mars.
```

Comment:
```suggestion
In order to have some more commits to compare, let's add a few more lines to Dracula's notes about Mars.  If you're working in your own project, and not the "planets" repo, you can still do hands-on work here. 
 Instead of using our suggestions for changes, try making small changes that won't break anything important, like adding an extra line between two paragraphs of text or correcting a typo in a code comment you've been meaning to get to.
```